### ISSUE TITLE
Get DataObject to automatically add the recommended index

### ISSUE THREAD

**Issue by sminnee** [silverstripe-australia/silverstripe-queuedjobs#300]
The docs suggest that you add this index:
```
ALTER TABLE `QueuedJobDescriptor` ADD INDEX ( `JobStatus` , `JobType` )
```

DataObject can be configured to add this index automatically with the `$indexes` static property. That would ensure that all installs receive this performance enhancement.

**Comment by mfendeksilverstripe** [comment]
There is already index on [JobStatus](https://github.com/symbiote/silverstripe-queuedjobs/blob/4.5/src/DataObjects/QueuedJobDescriptor.php#L95). I'm not sure if adding index on `JobType` will give much performance gain as cardinality of this index is really low. Note that `JobType` is actually `Queue type`. What we typically refer to as `type of job` is the `Implementation`. Adding index on this field would probably make more sense but I still think the cardinality of this index is too low as well.

**Comment by sminnee** [comment]
Right. I assumed that whoever wrote the doc (probably the original developer, @nyeholt) expected this index to be in place to the application to work well, but didn't investigate fully.
 
For my part I know that installations that don't have the job cleanup activated (and therefore get lots of completed jobs left in the system but hidden from the UI) end up with a very slow admin UI. I wondered if the absence of this index was the root cause.
